scipio/src/channels/local_channel.rs

```diff
@@ -0,0 +1,622 @@
+    /// It returns a [`ChannelError`] encapsulating a [`WouldBlock`] if this is a bounded channel that has no more capacity
+    ///
+    /// # Examples
+    /// ```
+    /// use scipio::{LocalExecutor, Local};
+    /// use scipio::channels::LocalChannel;
+    /// use futures::stream::StreamExt;
+    ///
+    /// let ex = LocalExecutor::make_default();
+    /// ex.run(async move {
+    ///     let (sender, mut receiver) = LocalChannel::new_bounded(1);
+    ///     sender.try_send(0);
+    ///     sender.try_send(0).unwrap_err(); // no more capacity
+    ///     receiver.next().await.unwrap(); // now we have capacity again
+    ///     drop(receiver); // but because the receiver is destroyed send will err
+    ///     sender.try_send(0).unwrap_err();
+    /// });
+    /// ```
+    ///
+    /// [`BrokenPipe`]: https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.BrokenPipe
+    /// [`Other`]: https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.Other
+    /// [`io::Error`]: https://doc.rust-lang.org/std/io/struct.Error.html
+    pub fn try_send(&self, item: T) -> Result<(), ChannelError<T>> {
+        self.channel.push(item)
+    }
+
+    /// Sends data into this channel when it is ready to receive it
+    ///
+    /// For an unbounded channel this is just a more expensive version of [`send`]. Prefer
+    /// to use [`send`] instead.
+    ///
+    /// For a bounded channel this will push to the channel when the channel is ready to
+    /// receive data.
+    ///
+    /// # Examples
+    /// ```
+    /// use scipio::{LocalExecutor, Local};
+    /// use scipio::channels::LocalChannel;
+    ///
+    /// let ex = LocalExecutor::make_default();
+    /// ex.run(async move {
+    ///     let (sender, receiver) = LocalChannel::new_bounded(1);
+    ///     sender.send(0).await.unwrap();
+    ///     drop(receiver);
+    /// });
+    /// ```
+    ///
+    /// [`send`]: struct.LocalSender.html#method.send
+    pub async fn send(&self, item: T) -> Result<(), ChannelError<T>> {
+        if self.has_room() {
+            self.try_send(item)
+        } else {
+            let waiter = SendWaiter {
+                channel: LocalChannel {
+                    state: self.channel.state.clone(),
+                },
+            };
+            waiter.await;
+            self.try_send(item)
+        }
+    }
+
+    /// Checks if there is room to send data in this channel
+    ///
+    /// # Examples
+    /// ```
+    /// use scipio::{LocalExecutor, Local};
+    /// use scipio::channels::LocalChannel;
+    ///
+    /// let ex = LocalExecutor::make_default();
+    /// ex.run(async move {
+    ///     let (sender, receiver) = LocalChannel::new_bounded(1);
+    ///     assert_eq!(sender.has_room(), true);
+    ///     sender.try_send(0);
+    ///     assert_eq!(sender.has_room(), false);
+    ///     drop(receiver);
+    /// });
+    /// ```
+    pub fn has_room(&self) -> bool {
```

Comment:
Alternative name would be is_full: https://docs.rs/flume/0.9.1/flume/struct.Sender.html#method.is_full